File: front/src/api/restaurant/dto/address.ts
import { z } from 'zod'

export const Address = z.object({
    streetNumber: z.number()
        .int()
        .positive({ message: "Street number must be a positive integer" }),

    streetName: z.string()
        .min(1, { message: "Street name is required" }),

    postalCode: z.string()
        .regex(/^\d{5}$/, { message: "Postal code must be 5 digits" }),

    city: z.string()
        .min(1, { message: "City is required" }),

    country: z.string()
        .default("France")
})

File: backend/src/main/kotlin/com/veg/bio/restaurant/domain/Address.kt
package com.veg.bio.restaurant.domain

data class Address(
    val streetNumber: Int,
    val streetName: String,
    val postalCode: String,
    val city: String,
    val country: String = "France"
){
    init {
        require(streetName.isNotBlank()) { "Le nom de la rue est obligatoire" }
        require(city.isNotBlank()) { "La ville est obligatoire" }
        require(postalCode.matches(Regex("\\d{5}"))) { "Code postal invalide" }
    }

    override fun toString(): String = "$streetNumber $streetName, $postalCode $city, $country"
}
